File: Light Code/VIQA/Common/Named.cs
using System;

namespace VIQA.Common
{
    public class Named
    {
        private string _name;

        protected Func<string> DefaultNameFunc = () => "";

        public string Name
        {
            set { _name = value; }
            get { return _name ?? DefaultNameFunc.Invoke(); }
        }
    }
}

File: Java/src/main/java/ru/viqa/ui_testing/common/Named.java
package ru.viqa.ui_testing.common;

import ru.viqa.ui_testing.common.funcInterfaces.FuncT;

/**
 * Created by roman.i on 24.09.2014.
 */
public class Named {
    private String _name;
    protected FuncT<String> DefaultNameFunc = () -> "";
    public String getName() throws Exception {
        return (_name != null) ? _name : DefaultNameFunc.invoke();
    }
    public void setName(String name) throws Exception { _name = name; }

}
